### PR TITLE
Control Add-In module to clean up JavaScript Add-Ins from AppDB

### PR DESCRIPTION
**Pull request by stkillen**
#### Summary 
Control Add-Ins should be coming with the system app and not as part of the `.bacpac` file. Introduce replacements for legacy control addins and create a proper structure and organise control add-ins in one module.

#### Work Item(s) 
Fixes [AB#492890](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/492890)

### CODE REVIEW

File: src/System Application/App/ControlAddIns/app.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+    "id":  "a3964a53-9345-458a-af20-d13f7eab7bd8",
+    "name":  "Control Add-Ins",
+    "publisher":  "Microsoft",
+    "brief":  "Provides the System control add-ins.",
+    "description":  "Provides the control add-in and their resources, used throughout the application.",
+    "version":  "23.0.0.0",
```
Comment (JesperSchulz) [inline src/System Application/App/ControlAddIns/app.json]: Should be 24.0

**Review by IhorHandziuk — DISMISSED** [review]

File: src/System Application/App/ControlAddIns/src/WebPageViewer/js/WebPageViewer.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
+/*! Copyright (C) Microsoft Corporation. All rights reserved. */
+
+var iframe = null;
+var subscribeToSrcLoad = function () { };
+var maxSize = 300;
+var defaultSize = '100%';
+var iframeHeight = defaultSize, iframeWidth = defaultSize;
+var pageTabSize = 20; // is 20 height of page tabs?
```
Comment (EmilDamsbo) [inline src/System Application/App/ControlAddIns/src/WebPageViewer/js/WebPageViewer.js]: Looks like a TODO left here

File: src/System Application/App/ControlAddIns/src/WebPageViewer/js/WebPageViewer.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    }
+
+    WebPageViewerHelper.TriggerCallback(s);
+  }
+
+  WebPageViewerHelper.SubscribeToEvent(eventName, recieveMessage);
+}
+
+function PostMessage(message, targetDomain, convertToJson) {
+  /// <summary>
```
Comment (EmilDamsbo) [inline src/System Application/App/ControlAddIns/src/WebPageViewer/js/WebPageViewer.js]: This docs comment seems like it should be above the method signature, or is that different in javascript files?

File: src/System Application/App/ControlAddIns/src/SatisfactionSurvey/SatisfactionSurvey.ControlAddin.al [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    HorizontalStretch = true;
+    HorizontalShrink = true;
+
+    Scripts = 'ControlAddIns\src\WebPageViewer\js\WebPageViewerHelper.js',
+              'ControlAddIns\src\WebPageViewer\js\WebPageViewer.js';
+    StartupScript = 'ControlAddIns\src\WebPageViewer\js\Startup.js';
+    RecreateScript = 'ControlAddIns\src\WebPageViewer\js\Recreate.js';
+    RefreshScript = 'ControlAddIns\src\WebPageViewer\js\Refresh.js';
+    StyleSheets = 'ControlAddIns\src\WebPageViewer\stylesheets\WebPageViewer.css';
+    Images = 'ControlAddIns\src\WebPageViewer\images\Callback.html', 'ControlAddIns\src\WebPageViewer\images\Loader.gif';
```
Comment (EmilDamsbo) [inline src/System Application/App/ControlAddIns/src/SatisfactionSurvey/SatisfactionSurvey.ControlAddin.al]: That seems hacky. We have an html document in an "Images" category, and the document only contains a single script tag for a callback - why not include it as just a script?

File: src/System Application/App/ControlAddIns/src/WelcomeWizard/WelcomeWizard.ControlAddin.al [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    /// <param name="GetStarted">Get started text</param>
+    /// <param name="GetStartedDescription">Description for the get started video</param>
+    /// <param name="GetHelp">Find Help Text</param>
+    /// <param name="GetHelpDescription">Description for the find help video</param>
+    /// <param name="RoleCenters">Role Centers Text</param>
+    /// <param name="RoleCentersDescription">Description about the role centers</param>
+    /// <param name="RoleCenter">Role center</param>
+    /// <param name="LegalDescription">Description explaining demo data is for demonstration purposes</param>
+    procedure Initialize(Title: Text; Subtitle: Text; Explanation: Text; Intro: Text; IntroDescription: Text; GetStarted: Text; GetStartedDescription: Text; GetHelp: Text;
+                         GetHelpDescription: Text; RoleCenters: Text; RoleCentersDescription: Text; RoleCenter: Text; LegalDescription: Text);
```
Comment (darjoo) [inline src/System Application/App/ControlAddIns/src/WelcomeWizard/WelcomeWizard.ControlAddin.al]: Weird additional line

File: src/System Application/App/ControlAddIns/src/WebPageViewer/js/WebPageViewer.js (hunk 2 above)
Comment (darjoo) [reply src/System Application/App/ControlAddIns/src/WebPageViewer/js/WebPageViewer.js]: Not created by Stig and seems to have been there forever. I'd suggest to just remove or leave the comment.

Comment (stkillen) [reply src/System Application/App/ControlAddIns/src/WebPageViewer/js/WebPageViewer.js]: Removed

**Review by EmilDamsbo — DISMISSED** [review]
Approving extension management changes

File: src/System Application/App/ControlAddIns/src/SatisfactionSurvey/SatisfactionSurvey.ControlAddin.al (hunk 4 above)
Comment (stkillen) [reply src/System Application/App/ControlAddIns/src/SatisfactionSurvey/SatisfactionSurvey.ControlAddin.al]: I agree, however the Satisfaction Survey Add-in is just 1-1 converted from the legacy add-in. I don't know if this was done for a specific reason and changing it around would affect the behaviour.

File: src/System Application/App/ControlAddIns/src/BusinessChart.ControlAddin.al [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    MinimumWidth = 200;
+    VerticalStretch = true;
+    VerticalShrink = true;
+    HorizontalStretch = true;
+    HorizontalShrink = true;
+
+    Scripts = 'Resources\BusinessChart\js\BusinessChartAddIn.js',
+              'https://ajax.aspnetcdn.com/ajax/jQuery/jquery-3.5.1.min.js',
+              'https://code.highcharts.com/9.1.1/highcharts.js',
+              'https://code.highcharts.com/9.1.1/highcharts-more.js',
```
Comment (pri-kise) [inline src/System Application/App/ControlAddIns/src/BusinessChart.ControlAddin.al:22]: How will those Control Add-Ins work in an OnPrem Scenario, that have limited access to any external ressources like CDNs?

And does this mean, that microsoft has aquired a license for highcharts and that we can use this library for custom control add-ins, too?

**Review by darjoo — APPROVED** [review]

File: src/System Application/App/ControlAddIns/src/BusinessChart.ControlAddin.al (hunk 6 above)
Comment (stkillen) [reply src/System Application/App/ControlAddIns/src/BusinessChart.ControlAddin.al:22]: This is a valid concern, I've created an internal work item to look into how to better accommodate OnPrem scenarios for these cases. Maybe these should be hosted locally or at least limited it to a single private CDN. For now, OnPrem environments with limited access to external resources will not benefit from these addins.

In regards to the license question I'm not too sure about that. This is a conversion of the old Business Chart control addin which was also using highcharts - so no new licenses have been acquired or any licensing changed as part of this PR. I'm not involved in the licensing part so please reach out through another channel if you want assistance with this inquiry.

**Review by JesperSchulz — APPROVED** [review]

**Review by WaelAbuSeada — COMMENTED** [review]
Reviewed updates to dotnet usage

**Review by WaelAbuSeada — APPROVED** [review]

**Pull request merged**